packages/cosmos/src/client.rs

```diff
@@ -147,6 +148,74 @@ impl<Res> PerformQueryWrapper<Res> {
 }
 
 impl Cosmos {
+    async fn get_and_update_simulation_sequence(
```

Comment:
I'm not sure we need the simulation sequence, but I could be mistaken. My theory is that the same node (or cluster of nodes) that we're requesting the current sequence number from will also be doing the simulation. The place where the problem really should be occurring is when the node we're simulating on disagrees with the current state of the chain (and therefore the current validator).

Maybe an important next step in this is modifying the error reporting for health reports to tell us whether the sequence mismatch occurred during the simulate or broadcast phase.

Reply:
I will create a new PR for updating the health report tomorrow to track where the failure is happening. Right now I think we have two options:
- Leave this as it is
- Remove this function

Don't have strong opinion either wise. What would you prefer ?

Reply:
We can leave it as-is for testing. We can deploy some testnet bots with all these changes and see what happens, it should be enlightening.